### PR TITLE
[EYE-117] Cloud Deployment

### PR DESCRIPTION
**Pull request by tucker-moore**
# Description
This PR adds a CDK package that'll manage provisioning our AWS deployment for us. Some actions are still manual (setting up database schema, adding IAM-enabled DB user), but those are one-time and I don't see any benefit in spending extra time to automate these as long as they're documented.

# Testing
The new assets are hosted at new subdomains (`auth2` and `api2`) and the mobile app has a new client ID. Use these when testing and deploying the mobile app.
- Logged in to app w/ Google, added a group, logged out + back in and the group is still there :)

### CODE REVIEW

File: server/cdk/test/cdk.test.ts [hunk 1]
```diff
@@ -0,0 +1,17 @@
+// import * as cdk from 'aws-cdk-lib';
+// import { Template } from 'aws-cdk-lib/assertions';
+// import * as Cdk from '../lib/cdk-stack';
+
+// example test. To run these tests, uncomment this file along with the
+// example resource in lib/cdk-stack.ts
+test('SQS Queue Created', () => {
+//   const app = new cdk.App();
```
Comment (1Revenger1) [inline server/cdk/test/cdk.test.ts:8]: Why is this all commented out? lol

File: server/cdk/assets/app.service [hunk 2]
```diff
@@ -0,0 +1,9 @@
+[Unit]
+Description=It's the backend
+Requires=network.target
+After=network.target
```
Comment (1Revenger1) [inline server/cdk/assets/app.service:4]: Is both After and Requires both needed?

**Review by 1Revenger1 — CHANGES_REQUESTED** [review]
Release github action needs to be changed to use OIDC_ENDPOINT too.

File: server/cdk/assets/app.service (hunk 2 above)
Comment (1Revenger1) [reply server/cdk/assets/app.service:4]: https://stackoverflow.com/questions/40593242/systemd-using-both-after-and-requires
Apparently both is good to have

File: server/cdk/test/cdk.test.ts (hunk 1 above)
Comment (tucker-moore) [reply server/cdk/test/cdk.test.ts:8]: That may or may not be the test that came with the starter project. I could update it to be snapshot tests for stability, but that might be annoying during development

**Review by matthew-latanafrancia — COMMENTED** [review]
ill approve on the github action chages

File: app/human-detector-app/app.config.js [hunk 3]
```diff
@@ -3,7 +3,7 @@ module.exports = {
     version: '1.0.0',
     "owner": "eyespy",
     extra: {
-      keycloakUrl: process.env.KEYCLOAK_URL,
```
[the remark this answers is not in the document]
Comment (tucker-moore) [reply app/human-detector-app/app.config.js:6]: o7

**Review by 1Revenger1 — APPROVED** [review]
Go go go
GitHub actions changes look good too

**Review by matthew-latanafrancia — APPROVED** [review]

**Pull request merged**